File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/verify/FormVerificationManager.java
package pl.edu.amu.wmi.erykandroidcommon.verify;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.support.annotation.NonNull;

import static android.text.TextUtils.isEmpty;

class FormVerificationManager {

    private final Context context;
    private final FieldVerifier fieldVerifier;

    FormVerificationManager(Context context, @NonNull FieldVerifier fieldVerifier) {
        this.context = context;
        this.fieldVerifier = fieldVerifier;
    }

    boolean verify() {
        VerificationResult verificationResult = fieldVerifier.verify();
        if (verificationResult != null) {
            if (!isEmpty(verificationResult.getErrorMessage())) {
                AlertDialog alertDialog = new AlertDialog.Builder(context).create();
                alertDialog.setMessage(verificationResult.getErrorMessage());
                alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, context.getString(android.R.string.ok), new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        dialogInterface.dismiss();
                    }
                });
                alertDialog.show();
                return false;
            } else if (verificationResult.isFailure()) {
                return false;
            }
        }
        return true;
    }


}

File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/verify/FormVerificationManager.kt
package pl.edu.amu.wmi.erykandroidcommon.verify

import android.app.AlertDialog
import android.content.Context

import android.text.TextUtils.isEmpty

internal class FormVerificationManager(private val context: Context, private val fieldVerifier: FieldVerifier) {

    fun verify(): Boolean {
        val verificationResult = fieldVerifier.verify()
        if (verificationResult != null) {
            if (!isEmpty(verificationResult.errorMessage)) {
                val alertDialog = AlertDialog.Builder(context).create()
                alertDialog.setMessage(verificationResult.errorMessage)
                alertDialog.setButton(AlertDialog.BUTTON_POSITIVE,
                        context.getString(android.R.string.ok)
                ) { dialogInterface, i -> dialogInterface.dismiss() }
                alertDialog.show()
                return false
            } else if (verificationResult.isFailure) {
                return false
            }
        }
        return true
    }


}
